ui/components/component-library/avatar-icon/avatar-icon.types.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  BackgroundColor,
+  BorderColor,
+  IconColor,
+  TextColor,
+} from '../../../helpers/constants/design-system';
+import { IconName, IconProps } from '../icon';
+import { AvatarBaseSize } from '../avatar-base/avatar-base.types';
+import { ValidTag } from '../text/text.types';
+import type { BoxProps } from '../../ui/box/box.d';
+
+export enum AvatarIconSize {
+  Xs = 'xs',
+  Sm = 'sm',
+  Md = 'md',
+  Lg = 'lg',
+  Xl = 'xl',
+}
+
+export interface AvatarIconProps extends BoxProps {
```

Comment:
Can we extend `AvatarBaseProps` here?

Reply:
```javascript
export interface AvatarIconProps extends AvatarBaseProps{
 ```
TS complains:
- Interface 'AvatarIconProps' incorrectly extends interface 'AvatarBaseProps'.
  Types of property 'size' are incompatible
-   Types of property 'color' are incompatible.

#### If we extend both
```javascript
export interface AvatarIconProps extends BoxProps, AvatarBaseProps{
 ```
TS complains:
- Named property 'children' of types 'BoxProps' and AvatarBaseProps  are not identical.

- Named property 'textAlign' of types 'BoxProps' and 'AvatarBaseProps' are not identical.